src/getExtension/getExtension.ts

```diff
@@ -7,22 +7,18 @@ import {
 
 const apiWindow = window as Window & ApiWindow
 
-function documentReadyPromise<T>(creator: () => T): Promise<T> {
-  return new Promise((resolve): void => {
-    if (document.readyState === 'complete') {
-      resolve(creator())
-    } else {
-      window.addEventListener('load', () => resolve(creator()))
-    }
-  })
-}
-
-export function getExtensions(): Promise<
-  Record<string, InjectedWindowProvider<PubSubSessionV1 | PubSubSessionV2>>
-> {
-  apiWindow.kilt = apiWindow.kilt || {}
+/**
+ * Get all extensions that are currently initialized and support the Credential API.
+ * 
+ * Note that this method only returns the extensions that are initialized at the time when this function is called.
+ * If an extension injects itself only after this function is called, it will not be contained in the returned extensions.
+ * @returns an object containing extensions
+ */
+export function getExtensions(): Record<string, InjectedWindowProvider<PubSubSessionV1 | PubSubSessionV2>> {
 
-  return documentReadyPromise(() => apiWindow.kilt)
+  // copy all extensions into a new object since the caller should be allowed to change the object
+  // without changing the underlying extension object.
+  return { ...apiWindow.kilt }
```

Comment:
I am surprised to see this requirement. This will also likely lose the very special versions property. I’d rather simply return the kilt object.

Reply:
As discussed, I changed it to an array. [0ea1196](https://github.com/KILTprotocol/kilt-extension-api/pull/11/commits/0ea1196b144b0ca1bfa68f61af1352cea0ba9271) and [f962370](https://github.com/KILTprotocol/kilt-extension-api/pull/11/commits/f9623709cc8085dfc07437fba82357efa6080819)